### PR TITLE
Fix csharp code in using_kinematic_body_2d.rst

### PR DESCRIPTION
**Pull request by berightben**
Match csharp code with gdscript code in the code-tab collider example.

### CODE REVIEW

**Comment by berightben** [comment]
I add another commit to fix typo.

**Comment by mhilbrunner** [comment]
@paulloz Any chance you can give this a look? :)

File: tutorials/physics/using_kinematic_body_2d.rst [hunk 1]
```diff
@@ -129,6 +129,23 @@ and ``get_slide_collision()``:
...
         var collision = get_slide_collision(i)
         print("I collided with ", collision.collider.name)
+        
+ .. code-tab:: csharp
+
+    // Using MoveAndCollide.
+    var collision = MoveAndCollide(velocity * delta);
+    if (collision != null)
+    {
+        GD.Print("I collided with ", collision.Collider.Name);
```
Comment (paulloz) [inline tutorials/physics/using_kinematic_body_2d.rst:139]:
```suggestion
        GD.Print("I collided with ", ((Node)collision.Collider).Name);
```

`KinematicCollision.Collider` is a `Godot.Object` which doesn't define `Name`.

File: tutorials/physics/using_kinematic_body_2d.rst [hunk 2]
```diff
@@ -129,6 +129,23 @@ and ``get_slide_collision()``:
...
+    }
+
+    // Using MoveAndSlide.
+    velocity = MoveAndSlide(velocity);
+    for (int i = 0; i < GetSlideCount(); i++)
+    {
+        var collision = GetSlideCollision(i);
+        GD.Print("I collided with ", collision.Collider.Name);
```
Comment (paulloz) [inline tutorials/physics/using_kinematic_body_2d.rst:147]:
```suggestion
        GD.Print("I collided with ", ((Node)collision.Collider).Name);
```

Same as above.

**Review by paulloz — CHANGES_REQUESTED** [review]

**Comment by paulloz** [comment]
Also, I'd prefer to see actual types (`KinematicCollision`) instead of `var` in these situations in the doc. As it's meant to be read on a website (and not in an IDE where you'd have Intellisense and all), it would be easier for the reader to know what type variables are.  
Though it doesn't feel right to block that particular PR for this, as `var` is used extensively in the docs already 🤔

**Comment by berightben** [comment]
> Also, I'd prefer to see actual types (`KinematicCollision`) instead of `var` in these situations in the doc. As it's meant to be read on a website (and not in an IDE where you'd have Intellisense and all), it would be easier for the reader to know what type variables are. Though it doesn't feel right to block that particular PR for this, as `var` is used extensively in the docs already thinking

I think it is good to mention both options available, maybe in the introduction of csharp. 
It is safe to go with what you recommended, to teach type safety. But maybe in another PR? Maybe C# maintainer can discussed it separately.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Pull request merged**